profiler.go

```diff
@@ -0,0 +1,19 @@
+//go:build profile
```

Comment:
Can you please move this out of root, somewhere like `pkg/utils` ?

Reply:
This will break the tag-based profiler enabling (ex:- local build with `go build -tags profile -o flagd` to produce flagd with profile). The reason is the usage of package "main" here.

Reply:
@AlexsJones  if this is not a strong preference, I will go ahead and resolve the conversation :)

Reply:
@james-milligan  yes that's correct. I thought about these options but they have many disadvantages compared to the current one. IMO, this usage is clean and confirms with use cases discussed by others [1]

[1] https://github.com/golang/go/issues/33389